frontend/src/components/UserMenu/AccountModal/index.jsx

```diff
@@ -34,6 +37,11 @@ export default function AccountModal({ user, hideModal }) {
     setPfp(null);
   };
 
+  const handleUsernameChange = (e) => {
+    const value = e.target.value.toLowerCase().replace(/\s/g, '');
```

Comment:
You can use the `pattern` attribute on inputs to do this sanitation on the form-level and it will block submission.